test/built-ins/String/prototype/split/limit-touint32-error.js

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2020 Richard Gibson. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-string.prototype.split
+description: Abrupt completion from ToUint32 on the limit
+info: |
+  1. If _limit_ is *undefined*, let _lim_ be 2<sup>32</sup> - 1; else let _lim_ be ? ToUint32(_limit_).
+  1. Let _R_ be ? ToString(_separator_).
+  1. If _lim_ = 0, return _A_.
+---*/
+
+var nonStringableSeparator = {};
+nonStringableSeparator[Symbol.toPrimitive] = $DONOTEVALUATE;
+nonStringableSeparator.toString = $DONOTEVALUATE;
+nonStringableSeparator.valueOf = $DONOTEVALUATE;
```

Comment:
`$DONOTEVALUATE` must only be used in tests that are explicitly a negative, parse phase Syntax Error test. The definition of `$DONOTEVALUATE` is allowed to be replaced by an engine's own runner, and that definition may not be what you want for these tests. 

For tests like this, just create your own function that throws whatever you want it to throw. If you don't have time or bandwidth, I will happily make the change for you

Reply:
@gibson042 I also want to acknowledge that this was never explicitly stated in the documentation for test contributing or interpreting and I apologize for that. I've updated the docs and I'm awaiting @leobalter's review.

Reply:
Thanks for the explanation. I've refactored to avoid violating such expectations. Should I also go a step further and generalize ExpectedError into the test harness and CONTRIBUTING.md?